src/Build.cmd

```diff
@@ -2,6 +2,38 @@
 setlocal
 
 SET CMDHOME=%~dp0.
+
+@REM Locate VS 2017 with the proper method
+
+SET VSWHERE_REMOTE_PATH=https://github.com/Microsoft/vswhere/releases/download/1.0.55/vswhere.exe
+SET VSWHERE_LOCAL_PATH=%CMDHOME%\vswhere.exe
+
+if NOT exist "%VSWHERE_LOCAL_PATH%" (
+  echo Downloading vswhere.exe from %VSWHERE_REMOTE_PATH%.
+  powershell -NoProfile -ExecutionPolicy unrestricted -Command "$retryCount = 0; $success = $false; do { try { (New-Object Net.WebClient).DownloadFile('%VSWHERE_REMOTE_PATH%', '%VSWHERE_LOCAL_PATH%'); $success = $true; } catch { if ($retryCount -ge 6) { throw; } else { $retryCount++; Start-Sleep -Seconds (5 * $retryCount); } } } while ($success -eq $false);"
+  if NOT exist "%VSWHERE_LOCAL_PATH%" (
+    echo ERROR: Could not install vswhere correctly.
+    exit /b 1
```

Comment:
With VS2017 there are NO env vars set for you, it is because they support SxS installations, so the RECOMMENDED way is to use vswhere to select the one you need. Later if we require to select a specific version we'll be able to add filtering criteria on vswhere output, which is the recommended way. So it is not a best practice anymore to depend on env vars (finally).